homeassistant/components/number/__init__.py

```diff
@@ -48,12 +48,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     component.async_register_entity_service(
         SERVICE_SET_VALUE,
         {vol.Required(ATTR_VALUE): vol.Coerce(float)},
-        "async_set_value",
+        async_set_value,
     )
 
     return True
 
 
+async def async_set_value(entity: NumberEntity, service_call: ServiceCall) -> None:
+    """Service call wrapper to set a new value."""
+    value = service_call.data["value"]
+    if value < entity.min_value or value > entity.max_value:
```

Comment:
We need to check and guard for min and max value being None here too.

Reply:
Nevermind, we have defaults in place and strict typing as float.

Reply:
Correct